### ISSUE TITLE
My post_load function seems to be called automatically: is this normal behavior?

### ISSUE THREAD

**Issue by nathanjwtx** [marshmallow-code/flask-smorest#489]
I have the following schema:
```
class UserRegisterSchema(UserSchema):
    email = fields.Str(required=True)

    @post_load
    def create_user(self, request_data, **kwargs):
        return UserModel(**request_data)
```
and the following part of a route:
```
@blp.route('/register')
class UserRegister(MethodView):
    @blp.arguments(UserRegisterSchema)
    def post(self, user_data):
        if UserModel.query.filter(or_(
                UserModel.email == user_data['email'],
                UserModel.username == user_data['username'])).first():
            abort(409, message='That username already exists')

        user = user_regiser_schema.load(user_data)
```

With the `@post_load` decorator in place on the Schema I get a TypeError saying `user_data['email']` is not subscritable. If I comment out the `@post_load` function the route works as expected. So it appears that the `@post_load` function is being called automatically thereby turning `user_data` into an object rather than leaving it is as the expected dictionary.

Is this something that flask-smorest does? I haven't encountered this when using regular Flask and marshmallow.

**Comment by lafrech** [comment]
This is something marshmallow does. What flask-smorest does is just call the deserialize function. It is the point of post_load to be called upon deserialization.

[Here](https://github.com/BEMServer/bemserver-api/blob/master/bemserver_api/resources/campaigns/routes.py)'s an example of how I typically structure this kind of CRUD route. Indeed, I don't instantiate an object in the schema. While it might be a good idea in a POST request, it wouldn't make sense for a PUT.

Bottom line, you're probably better off removing the post-load.

As a sidenote, it is considered better practice to commit and catch the unique constraint error than to query before posting a new instance. If only to protect against race conditions. In the repo linked above, this is what I do with a custom `catch_integrity_error` decorator.
